src/Uno.UI/UI/Xaml/Application.macOS.cs

```diff
@@ -87,7 +87,15 @@ public override void DidFinishLaunching(NSNotification notification)
 			}
 			if (!handled)
 			{
-				OnLaunched(new LaunchActivatedEventArgs());
+				var argumentsString = "";
+				
+				// First argument is just executable path
+				if (NSProcessInfo.ProcessInfo?.Arguments is { Length: > 1} processArguments)
```

Comment:
this check is unnecessary, you are checking for that in `GetCommandLineArgsWithoutExecutable`